### PR TITLE
Only make composition suggestion on `object . app function`

### PR DESCRIPTION
**Pull request by mjgpy3**
Seems to fix the cases given by @hdgarrood in #1272. I'm not sure it's totally done, but is more restrictive, I'd like to throw more cases at it if someone can come up some that are interesting.

### CODE REVIEW

**Comment by mjgpy3** [comment]
(removed a debug trace)

**Comment by mjgpy3** [comment]
Here are the outputs

SuggestComposition.purs:
```
Writing output/Prelude/index.js
Writing output/Prelude/foreign.js
Writing output/Prelude/externs.purs
Compiling SuggestComposition
Error:
Error in module SuggestComposition:
Error in value declaration f:
Error checking that type
  _0 -> _1
subsumes type
  { g :: _2 | _3 }
Error checking that type
  { g :: _2 | _3 }
subsumes type
  _0 -> _1
Error at /home/michael/dev/contrib/purescript/examples/failing/SuggestComposition.purs line 5, column 5 - line 5, column 11:
  Cannot unify type
    Prim.Object
  with type
    Prim.Function _0
Note that function composition in PureScript is defined using (<<<)
See https://github.com/purescript/purescript/wiki/Error-Code-TypesDoNotUnify for more information, or to contribute content related to this error.
```

DoNotSuggestComposition.purs:
```
Compiling DoNotSuggestComposition
Error:
Error in module DoNotSuggestComposition:
Error in value declaration bar:
Error checking that type
  { y :: Prim.Int }
subsumes type
  Prim.String
Error at /home/michael/dev/contrib/purescript/examples/failing/DoNotSuggestComposition.purs line 10, column 7 - line 10, column 11:
  Cannot unify type
    { y :: Prim.Int }
  with type
    Prim.String
See https://github.com/purescript/purescript/wiki/Error-Code-TypesDoNotUnify for more information, or to contribute content related to this error.
```

DoNotSuggestComposition2.purs:
```
Compiling DoNotSuggestComposition2
Error:
Error in module DoNotSuggestComposition2:
Error in value declaration foo:
Error at /home/michael/dev/contrib/purescript/examples/failing/DoNotSuggestComposition2.purs line 3, column 27 - line 3, column 29:
  Cannot apply function of type
    { y :: Prim.Int }
  to argument
    2
See https://github.com/purescript/purescript/wiki/Error-Code-CannotApplyFunction for more information, or to contribute content related to this error.
```

File: src/Language/PureScript/Errors.hs [hunk 1]
```diff
@@ -670,7 +670,7 @@ prettyPrintSingleError full level e = prettyPrintErrorMessage <$> onTypesInError
                                              , indent . line $ "import " ++ show im ++ " hiding (" ++ nm ++ ")"
                                              ]
     suggestions' (TypesDoNotUnify t1 t2)
-      | any isObject [t1, t2] = [line "Note that function composition in PureScript is defined using (<<<)"]
+      | isObject t1 && isFunction t2 = [line "Note that function composition in PureScript is defined using (<<<)"]
```
Comment (paf31) [inline src/Language/PureScript/Errors.hs]: We need to test the other way round too, don't we?

**Comment by paf31** [comment]
Thanks for working on this!

File: src/Language/PureScript/Errors.hs (hunk 1 above)
Comment (mjgpy3) [inline src/Language/PureScript/Errors.hs]: Honestly, I'm not really sure. Can you give me a code example of what that might look like?

Comment (mjgpy3) [inline src/Language/PureScript/Errors.hs]: As in, a _PureScript example_

**Comment by mjgpy3** [comment]
@paf31 thanks!

**Pull request merged**